include/internal/refcount.h

```diff
@@ -12,11 +12,8 @@
 
 # include <openssl/e_os2.h>
 
-/* Used to checking reference counts, most while doing perl5 stuff :-) */
-# if defined(OPENSSL_NO_STDIO)
-#  if defined(REF_PRINT)
-#   error "REF_PRINT requires stdio"
-#  endif
+# if defined(REF_DEBUG) &&  defined(OPENSSL_NO_STDIO)
+#  error "REF_DEBUG requires stdio"
 # endif
```

Comment:
I forgot...  you can drop this check entirely as well.  Now that you have made the REF_PRINT macros to use OSSL_TRACE macros, which do not depend on stdio, this condition is simply not true any more.